### PR TITLE
[FEAT] Plaza Username

### PR DESCRIPTION
**Pull request by 0xSacul**
# Description

This PR adds a way to display a custom username in Plaza instead of your Farm ID (still wip)

image

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

# Checklist:

- [x] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [x] Screenshot if it includes any UI changes
- [x] I have read the contributing guidelines and agree to the T&Cs
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by 0xSacul** [comment]
eyo

**Comment by 0xSacul** [comment]
## Warning

Do not merge before back-end updates, this PR requires changes on Colyseus server to handle username.

class Player need `@type("string") username?: string;`
message & player need username string.

This PR also needs some stress test before merge (pls)

**Review by c-py — APPROVED** [review]
LGTM!

**Pull request merged**
